File: ruby/raindrops/raindrops.rb
module BookKeeping
  VERSION = 3
end

class Raindrops
  RULES = {
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong'
  }

  def Raindrops.convert(drop)
    sounds = RULES.sort.each_with_object '' do |(n, sound), result|
      result << sound if drop % n == 0
    end

    sounds.empty? ? drop.to_s : sounds
  end
end

File: nim/raindrops/raindrops.nim
import future, sequtils, strutils

const rules: seq[tuple[factor: int, sound: string]] =
  @[(3, "Pling"), (5, "Plang"), (7, "Plong")]

proc convert*(n: int): string =
  let sounds = lc[ rule.sound | ( rule <- rules, n %% rule.factor == 0 ),
                   string ]
  if len(sounds) > 0:
    sounds.join("")
  else:
    $n

